src/Raven.Server/Documents/Replication/ConflictManager.cs

```diff
@@ -171,15 +173,33 @@ public static void AssertChangeVectorNotNull(string conflictedChangeVector)
 
             conflictedDocs.Add(conflict.Clone());
             conflictedDocs.Sort((x, y) => string.Compare(x.ChangeVector, y.ChangeVector, StringComparison.Ordinal));
-
-            if (_conflictResolver.TryResolveConflictByScriptInternal(
-                documentsContext,
-                scriptResolver,
-                conflictedDocs,
-                documentsContext.GetLazyString(collection), out var resolved))
+            try
             {
-                _conflictResolver.PutResolvedDocument(documentsContext, resolved, resolvedToLatest: false, conflict);
-                return true;
+                if (_conflictResolver.TryResolveConflictByScriptInternal(
+                        documentsContext,
+                        scriptResolver,
+                        conflictedDocs,
+                        documentsContext.GetLazyString(collection), out var resolved))
+                {
+                    _conflictResolver.PutResolvedDocument(documentsContext, resolved, resolvedToLatest: false, conflict);
+                    return true;
+                }
+            }
+            catch (Exception e)
```

Comment:
can we catch that inside `TryResolveConflictByScriptInternal` ?
because I see there is one other place that use `TryResolveConflictByScriptInternal` and we want to have same behavior